packages/dashboard/src/components/Wizard/examples/SetupWizard.Util.ts

```diff
@@ -0,0 +1,222 @@
+import { ISubwayNavStep, SubwayNavStepState } from '../../SubwayNav/SubwayNav.types';
+import { IWizardStepProps } from '../Wizard.types';
+
+export interface ISetupWizardState {
+  currStep: IWizardStepProps;
+  currSubStep: IWizardStepProps | undefined;
+  steps: IWizardStepProps[];
+}
+
+/**
+ * generate Random id
+ */
+export function generateRandomId(): string {
+  return (
+    Math.random()
+      .toString(36)
+      .substring(2) + new Date().getTime().toString(36)
+  );
+}
+
+// Go to a given parent step -> substep
+export function goToStep(
+  currentState: ISetupWizardState,
+  parentStepToGo: ISubwayNavStep,
+  subStepToGo: ISubwayNavStep | undefined
+): ISetupWizardState | undefined {
+  if (parentStepToGo === undefined) {
+    // do nothing
+    return;
+  }
+
+  if (parentStepToGo.key === currentState.currStep.key) {
+    // Already in current step, go to Sub Step
+    if (subStepToGo !== undefined) {
+      return goToSubStep(currentState, parentStepToGo, subStepToGo);
+    }
+  } else {
+    let newSteps: IWizardStepProps[] = [];
+    newSteps = currentState.steps;
+
+    let currStep = currentState.currStep;
+    let currSubStep = currentState.currSubStep;
+
+    let foundStepToGo: boolean = false;
+
+    newSteps.map((stepObj: IWizardStepProps) => {
+      if (stepObj.key === currentState.currStep.key) {
+        stepObj.state = SubwayNavStepState.Completed; /// Only for testing (get state from wizard content state?)
+
+        if (stepObj.subSteps !== undefined && stepObj.subSteps.length > 0) {
+          stepObj.subSteps.forEach((obj: IWizardStepProps) => {
+            obj.state = SubwayNavStepState.Completed; /// Only for testing (get state from wizard content state?)
+          });
+
+          // Since current step is completed, reset current sub step
+          currSubStep = undefined;
+        }
+      } else if (stepObj.key === parentStepToGo.key) {
+        stepObj.state = SubwayNavStepState.Current;
+        currStep = stepObj;
+
+        if (stepObj.subSteps !== undefined && stepObj.subSteps.length > 0) {
+          stepObj.subSteps[0].state = SubwayNavStepState.Current;
+          currSubStep = stepObj.subSteps[0];
+        }
+        foundStepToGo = true;
+      } else if (!foundStepToGo && stepObj.state === SubwayNavStepState.NotStarted) {
+        stepObj.state = SubwayNavStepState.Skipped;
+      }
+    });
+
+    if (foundStepToGo) {
+      // this.setState({ steps: newSteps, currStep: currStep, currSubStep: currSubStep });
+      return { steps: newSteps, currStep: currStep, currSubStep: currSubStep };
+    }
+  }
+}
+
+export function goToSubStep(
```

Comment:
I felt this can be the same as the above function, if you take
* the root NavStep.
* the parentKey
* the subKey

Reply:
Just test code.

Reply:
Oh I see, so the same utility is not available to use by users yet.